datahub-web-react/src/app/entity/user/UserEditProfileModal.tsx

```diff
@@ -0,0 +1,242 @@
+    onCreate: () => void;
+    data: {
+        name: string | undefined;
+        title: string | undefined;
+        image: string | undefined;
+        team: string | undefined;
+        email: string | undefined;
+        slack: string | undefined;
+        phone: string | undefined;
+        urn: string | undefined;
+    };
+};
+/** Regex Validations */
+export const validUserName = new RegExp('^[a-zA-Z ]*$');
+
+export default function UserEditProfileModal({ visible, onClose, onCreate, data }: Props) {
+    const [updateCorpUserPropertiesMutation] = useUpdateCorpUserPropertiesMutation();
+    const [form] = Form.useForm();
+
+    const [userName, setUserName] = useState<string | undefined>('');
+    const [userTitle, setUserTitle] = useState<string | undefined>('');
+    const [userImageURL, setImageURL] = useState<string | undefined>('');
+    const [userTeam, setUserTeam] = useState<string | undefined>('');
+    const [userEmail, setUserEmail] = useState<string | undefined>('');
+    const [userSlack, setUserSlack] = useState<string | undefined>('');
+    const [userPhoneNumber, setUserPhoneNumber] = useState<string | undefined>('');
+    const [buttonDisabled, setButtonDisabled] = useState(true);
+
+    useEffect(() => {
+        if (data) {
+            setUserName(data.name);
+            setUserTitle(data.title);
+            setImageURL(data.image);
+            setUserTeam(data.team);
+            setUserEmail(data.email);
+            setUserSlack(data.slack);
+            setUserPhoneNumber(data.phone);
+        }
+    }, [data]);
+
+    // save changes function
+    const onCreateGroup = () => {
+        updateCorpUserPropertiesMutation({
+            variables: {
+                urn: data?.urn || '',
+                input: {
+                    displayName: userName,
+                    title: userTitle,
+                    pictureLink: userImageURL,
+                    teams: userTeam?.split(','),
+                    email: userEmail,
+                    slack: userSlack,
+                    phone: userPhoneNumber,
+                },
+            },
+        })
+            .catch((e) => {
+                message.destroy();
+                message.error({ content: `Failed to Save changes!: \n ${e.message || ''}`, duration: 3 });
+            })
+            .finally(() => {
+                message.success({
+                    content: `Changes saved.`,
+                    duration: 3,
+                });
+                onCreate();
+                setUserName('');
+                setUserTitle('');
+                setImageURL('');
+                setUserTeam('');
+                setUserEmail('');
+                setUserSlack('');
+                setUserPhoneNumber('');
+            });
+        onClose();
+        onCreate();
+    };
+
+    // userFieldValidation before submitting the changes.
+    const userFieldValidations = () => {
+        if ((userName !== '' || data.name) && !buttonDisabled) {
```

Comment:
can you explain this? this is saying the user field is invalid if userName is not blank or data.name is provided and the button is not disabled? that doesn't make intuitive sense to me

Reply:
Yeah, the other validation were added earlier, it skip from my mind to remove the logic. 
I have added the form validations as discussed in the last call, that would be sufficient to validate all the fields. 
Removing the other and only keeping 'buttonDisabled'.